### PR TITLE
Tcp and UDP connection specs failure Java 11

### LINKED ISSUE #25733: FAILED akka.io.TcpConnectionSpec on Java 11
There seems to be issues around closing connections, I see UDP tests failing with port already used errors locally also:

https://jenkins.akka.io:8498/job/akka-nightly-jdk11/16/consoleFull

Some of the failures seen locally:
```
[ERROR] [10/04/2018 09:24:18.303] [UdpConnectedIntegrationSpec-akka.actor.default-dispatcher-7] [akka://UdpConnectedIntegrationSpec/system/IO-UDP-FF/selectors/$a/3] Failed to bind UDP channel to endpoint [/127.0.0.1:65070]
java.net.BindException: Address already in use
...
[info] - must be able to unbind and bind again successfully *** FAILED *** (48 milliseconds)
[info]   java.lang.AssertionError: assertion failed: expected Bound(/127.0.0.1:65070), found CommandFailed(Bind(Actor[akka://UdpConnectedIntegrationSpec/system/testActor-497#-1328237810],/127.0.0.1:65070,List()))
[info]   at scala.Predef$.assert(Predef.scala:219)
[info]   at akka.testkit.TestKitBase.expectMsg_internal(TestKit.scala:403)
[info]   at akka.testkit.TestKitBase.expectMsg(TestKit.scala:379)
[info]   at akka.testkit.TestKitBase.expectMsg$(TestKit.scala:379)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:896)
[info]   at akka.io.UdpConnectedIntegrationSpec.bindUdp(UdpConnectedIntegrationSpec.scala:23)
[info]   at akka.io.UdpConnectedIntegrationSpec.$anonfun$new$4(UdpConnectedIntegrationSpec.scala:88)
```

```
info] - must abort the connection and reply with `Aborted` upon reception of an `Abort` command *** FAILED *** (172 milliseconds)
[info]   Expected exception java.io.IOException to be thrown, but no exception was thrown (TcpConnectionSpec.scala:459)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.MatchersHelper$.newTestFailedException(MatchersHelper.scala:138)
[info]   at org.scalatest.MatchersHelper$.checkExpectedException(MatchersHelper.scala:322)
[info]   at org.scalatest.words.ResultOfTheTypeInvocation.thrownBy(ResultOfTheTypeInvocation.scala:39)
[info]   at akka.io.TcpConnectionSpec$$anon$18.$anonfun$new$41(TcpConnectionSpec.scala:459)
[info]   at akka.io.TcpConnectionSpec$EstablishedConnectionTest.$anonfun$run$2(TcpConnectionSpec.scala:992)
[info]   at akka.io.TcpConnectionSpec$UnacceptedConnectionTest.$anonfun$run$1(TcpConnectionSpec.scala:946)
[info]   at akka.io.TcpConnectionSpec$LocalServerTest.run(TcpConnectionSpec.scala:895)
[info]   at akka.io.TcpConnectionSpec$UnacceptedConnectionTest.run(TcpConnectionSpec.scala:943)
[info]   at akka.io.TcpConnectionSpec$EstablishedConnectionTest.run(TcpConnectionSpec.scala:978)
[info]   at akka.io.TcpConnectionSpec$$anon$18.<init>(TcpConnectionSpec.scala:452)
[info]   at akka.io.TcpConnectionSpec.$anonfun$new$40(TcpConnectionSpec.scala:451)
```

```
[info] - must report when peer aborted the connection *** FAILED *** (3 seconds, 73 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.io.Tcp$ErrorClosed
[info]   at scala.Predef$.assert(Predef.scala:219)
[info]   at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:509)
[info]   at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:482)
[info]   at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:482)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:896)
[info]   at akka.io.TcpConnectionSpec$$anon$22.$anonfun$new$54(TcpConnectionSpec.scala:596)
[info]   at akka.io.TcpConnectionSpec$EstablishedConnectionTest.$anonfun$run$2(TcpConnectionSpec.scala:992)
[info]   at akka.io.TcpConnectionSpec$UnacceptedConnectionTest.$anonfun$run$1(TcpConnectionSpec.scala:946)
[info]   at akka.io.TcpConnectionSpec$LocalServerTest.run(TcpConnectionSpec.scala:895)
[info]   at akka.io.TcpConnectionSpec$UnacceptedConnectionTest.run(TcpConnectionSpec.scala:943)
[info]   at akka.io.TcpConnectionSpec$EstablishedConnectionTest.run(TcpConnectionSpec.scala:978)
[info]   at akka.io.TcpConnectionSpec$$anon$22.<init>(TcpConnectionSpec.scala:593)
[info]   at akka.io.TcpConnectionSpec.$anonfun$new$53(TcpConnectionSpec.scala:592)
```

```
[info] - must report when peer closed the connection when trying to write *** FAILED *** (3 seconds, 77 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (3 seconds) during expectMsgClass waiting for class akka.io.Tcp$ErrorClosed
[info]   at scala.Predef$.assert(Predef.scala:219)
[info]   at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:509)
[info]   at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:482)
[info]   at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:482)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:896)
[info]   at akka.io.TcpConnectionSpec$$anon$23.$anonfun$new$56(TcpConnectionSpec.scala:613)
[info]   at akka.io.TcpConnectionSpec$EstablishedConnectionTest.$anonfun$run$2(TcpConnectionSpec.scala:992)
[info]   at akka.io.TcpConnectionSpec$UnacceptedConnectionTest.$anonfun$run$1(TcpConnectionSpec.scala:946)
[info]   at akka.io.TcpConnectionSpec$LocalServerTest.run(TcpConnectionSpec.scala:895)
[info]   at akka.io.TcpConnectionSpec$UnacceptedConnectionTest.run(TcpConnectionSpec.scala:943)
[info]   at akka.io.TcpConnectionSpec$EstablishedConnectionTest.run(TcpConnectionSpec.scala:978)
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #25733

Would be good with some @jrudolph eyes on this. I think I fixed it but confidence level is not 100. 

Tested on Java 11 and 8, but not on Windows (which might also make sense to do)

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM, if tested thoroughly, including windows

File: akka-actor/src/main/scala/akka/io/UdpListener.scala [hunk 1]
```diff
@@ -77,7 +77,9 @@ private[io] class UdpListener(
       log.debug("Unbinding endpoint [{}]", bind.localAddress)
       try {
         channel.close()
-        if (Helpers.isWindows) registration.enableInterest(OP_READ)
+        registration.enableInterest(OP_READ)
```
Comment (chbatey) [inline akka-actor/src/main/scala/akka/io/UdpListener.scala]: is this intended? Should it just be set in the if statement below?

Comment (johanandren) [reply akka-actor/src/main/scala/akka/io/UdpListener.scala]: Ups. Should only be below, duplicated now

Comment (2m) [reply akka-actor/src/main/scala/akka/io/UdpListener.scala]: Should I test on Windows JDK11 with this removed?

Comment (johanandren) [reply akka-actor/src/main/scala/akka/io/UdpListener.scala]: I don't think that matters, it will just enable the same interest twice.

**Comment by johanandren** [comment]
Let's close this and if there is anything to salvage (I don't think so) @jrudolph can pick that up in a new PR

**Comment by jrudolph** [comment]
Yep, I'll follow up on this. For `TcpConnectionSpec`, it seems we indeed need something very close to your fix. In the end, I had to step through the JDK 11 code with a debugger to figure out what was going on...

**Pull request closed without merging**